File: certification_server/src/iface/common/key_manager.go
package common

import (
	"context"

	commsecmodel "certification_server/src/models/commsec"
)

// IKeyManager 定义密钥管理相关操作。
type IKeyManager interface {
	// GetPublicKey 获取本地服务的公钥信息。
	GetPublicKey(ctx context.Context) (commsecmodel.ServicePublicKeyRecord, error)
	// GetPrivateKeyRef 获取本地服务的私钥引用信息。
	GetPrivateKeyRef(ctx context.Context) (commsecmodel.LocalPrivateKeyRef, error)
	// 通过指定信息查询公钥记录，支持多维度查询条件（如实体ID、密钥ID等）。
	LookupPublicKey(ctx context.Context, req *commsecmodel.PublicKeyLookupRequest) (commsecmodel.PublicKeyLookupResult, error)
}

File: data_server/src/iface/common/key_manager.py
from abc import ABC, abstractmethod

from src.models.commsec.commsec import (
    LocalPrivateKeyRef,
    PublicKeyLookupRequest,
    PublicKeyLookupResult,
    ServicePublicKeyRecord,
)


class IKeyManager(ABC):
    """本地密钥管理和公钥目录查询接口，提供本地公私钥访问和目录查询功能。"""

    @abstractmethod
    async def get_public_key(self) -> ServicePublicKeyRecord:
        """获取本地服务当前使用的公钥记录。"""
        raise NotImplementedError

    @abstractmethod
    async def get_private_key_ref(self) -> LocalPrivateKeyRef:
        """获取活动的本地私钥引用。"""
        raise NotImplementedError

    @abstractmethod
    async def lookup_public_key(
        self,
        req: PublicKeyLookupRequest,
    ) -> PublicKeyLookupResult:
        """根据 key_id, entity_id 或 owner 维度查找目录密钥。"""
        raise NotImplementedError

